### DISCUSSION TITLE
It would give an "Unsupported media type" error when making POST request with empty body

### DISCUSSION THREAD

**Discussion by MetinTAPAN58** [ThreeMammals/Ocelot#1864] [Show and tell]
## Expected Behavior / New Feature

If the request body is null in a POST request the process of  must continue

## Actual Behavior / Motivation for New Feature

If the request body is null in a POST request or "Content-Length: 0" is given as the Header parameter, it would give an "Unsupported media type" error when making the request over the ApiGateway.
  
## Specifications 
  - Version: 13.5.2 

##  To fixed it my solution: 
  
`Ocelot/Request/Mapper/`  
    
Changes will be required to the [MapContent](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/Request/Mapper/RequestMapper.cs#L28) method in [RequestMapper.cs](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/Request/Mapper/RequestMapper.cs).
I edited the code to avoid error in POST requests if request body is empty

```csharp
     private async Task<HttpContent> MapContent(HttpRequest request)
     { 
         var breakLine = false;
         if (request.Body == null || (request.Body.CanSeek && request.Body.Length <= 0))
         {
             breakLine = true;
         }

         // Never change this to StreamContent again, I forgot it doesnt work in #464.
         var content = new ByteArrayContent(await ToByteArray(request.Body));

         if (!string.IsNullOrEmpty(request.ContentType))
         {
             content.Headers
                 .TryAddWithoutValidation("Content-Type", new[] { request.ContentType });
         }

         if(breakLine)
         {
             return content;
         }

         AddHeaderIfExistsOnRequest("Content-Language", content, request);
         AddHeaderIfExistsOnRequest("Content-Location", content, request);
         AddHeaderIfExistsOnRequest("Content-Range", content, request);
         AddHeaderIfExistsOnRequest("Content-MD5", content, request);
         AddHeaderIfExistsOnRequest("Content-Disposition", content, request);
         AddHeaderIfExistsOnRequest("Content-Encoding", content, request);

         return content;
     }
```

**Comment by PrinceTegaton** [comment]
I think its best for the downstream service to handle with its default / configured behavior. Ocelot will be better as just a pipeline with less interfering logic

**Comment by raman-m** [comment]
Hi Metin!
Request Mapper is critical logic of the gateway ❗ 
**We are not going to change anything that you've proposed.**
But you can discuss this user scenario asking help from the community.

**Reply by raman-m** [reply]
@ggnaegi  FYI
He wanted to break our holy logic! :scream_cat:
